src/main/java/nl/tudelft/model/bubble/Bubble.java

```diff
@@ -31,8 +31,8 @@
     private boolean frozen = false;
     private int tickCount = 0;
     private final ResourcesWrapper resources;
-    private final List<Bubble> next;
-    private final boolean goesRightInitially;
+    protected BubbleFactory bubbleFactory = null;
```

Comment:
I disagree, the children <b>must</b> set that field manually (by setting it with the correct factory), otherwise the class contract is broken. They therefore must touch that field, i.e. they use it as one of their own fields. This is unlike other private superclass attributes, which are only used and set by the superclass' methods / constructor.

< interface